src/sentry/api/validators/sentry_apps/schema.py

```diff
@@ -0,0 +1,261 @@
+
+        'fieldset': {
+            'type': 'array',
+            'minItems': 1,
+            'items': [
+                {
+                    'anyOf': [
+                        {'$ref': '#/definitions/select'},
+                        {'$ref': '#/definitions/text'},
+                    ],
+                },
+            ],
+        },
+
+        # Composable Components
+
+        'select': {
+            'type': 'object',
+            'properties': {
+                'type': {
+                    'type': 'string',
+                    'enum': ['select'],
+                },
+                'label': {
+                    'type': 'string',
+                },
+                'name': {
+                    'type': 'string',
+                },
+                'uri': {
+                    '$ref': '#/definitions/uri',
+                },
+                'options': {
+                    '$ref': '#/definitions/options',
+                },
+            },
+            'required': ['type', 'name', 'label'],
+            'oneOf': [
+                {'required': ['uri']},
+                {'required': ['options']},
+            ],
+        },
+
+        'text': {
+            'type': 'object',
+            'properties': {
+                'type': {
+                    'type': 'string',
+                    'enum': ['text'],
+                },
+                'label': {
+                    'type': 'string',
+                },
+                'name': {
+                    'type': 'string',
+                },
+            },
+            'required': ['type', 'label', 'name'],
+        },
+
+        'header': {
+            'type': 'object',
+            'properties': {
+                'type': {
+                    'type': 'string',
+                    'enum': ['header'],
+                },
+                'text': {
+                    'type': 'string',
+                },
+            },
+            'required': ['type', 'text'],
+        },
+
+        'image': {
+            'type': 'object',
+            'properties': {
+                'type': {
+                    'type': 'string',
+                    'enum': ['image'],
+                },
+                'url': {
+                    'type': 'string',
+                    'format': 'uri',
+                    'pattern': '^https?://',
+                },
+                'alt': {
+                    'type': 'string',
+                },
+            },
+            'required': ['type', 'url'],
+        },
+
+        'video': {
+            'type': 'object',
+            'properties': {
+                'type': {
+                    'type': 'string',
+                    'enum': ['video'],
+                },
+                'url': {
+                    'type': 'string',
+                    'format': 'uri',
+                    'pattern': '^https?://',
+                },
+            },
+            'required': ['type', 'url'],
+        },
+
+        'markdown': {
```

Comment:
Markdown generally allows arbitrary HTML to be used. Be careful.

Reply:
Yeaaaaaaa, we were talking about this this morning. This whole thing might need to be in an `iframe` as well.

Reply:
You could also run something like [bleach](https://pypi.org/project/bleach/) over the rendered HTML allowing you to safely subset both markdown and HTML.